unstructured/partition/text_type.py

```diff
@@ -283,16 +291,33 @@ def exceeds_cap_ratio(text: str, threshold: float = 0.5) -> bool:
     # ex. world_tokenize("ITEM 1. Financial Statements (Unaudited)")
     #     = ['ITEM', '1', '.', 'Financial', 'Statements', '(', 'Unaudited', ')'],
     # however, "ITEM 1. Financial Statements (Unaudited)" is Title, not NarrativeText
-    tokens = [tk for tk in word_tokenize(text) if tk.isalpha()]
+    # tokens = [tk for tk in word_tokenize(text) if tk.isalpha()]
+    tokens = word_tokenize(text)
+    is_alpha = np.array([token.isalpha() for token in tokens])
 
-    # NOTE(jay-ylee) - If word_tokenize(text) is empty, return must be True to
-    # avoid being misclassified as Narrative Text.
-    if len(tokens) == 0:
+    # NOTE(klaijan) - If all alphas in word_tokenize(text) is empty
+    # return must be True to avoid being misclassified as Narrative Text.
+    if sum(is_alpha) == 0:
         return True
 
-    capitalized = sum([word.istitle() or word.isupper() for word in tokens])
-    ratio = capitalized / len(tokens)
-    return ratio > threshold
+    is_capitalized = np.array([word.istitle() or word.isupper() for word in tokens])
+    is_ne = np.zeros(len(tokens))
+    i = 0
+    for chunk in ne_chunk(text):
+        if hasattr(chunk, "label"):
+            for word in range(len(chunk)):
+                is_ne[word + i] = 1
+        i += 1
+
+    ignored_capitalize = sum(np.logical_and(is_alpha, np.logical_or(is_capitalized, is_ne)))
+
+    if sum(is_alpha) > ignored_capitalize:
+        ratio = (sum(np.logical_and(is_alpha, is_capitalized)) - ignored_capitalize) / (
+            sum(is_alpha) - ignored_capitalize
+        )
+    else:
+        ratio = sum(np.logical_and(is_alpha, is_capitalized)) / sum(is_alpha)
+    return bool(ratio > threshold)
```

Comment:
devil's advocate (for maybe a followup ticket): since `threshold` is used only in the end as gatecheck maybe we want to refactor this so there is a function that just returns the ratio just in case in the future we may want to use the ratio for other purposes. This also makes the function logically more modular (just computes the ratio of capitalized letters).